app/main.py

```diff
@@ -1,8 +1,26 @@
 class Person:
-    # write your code here
-    pass
 
+    people = {}
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
+
+
+def create_person_list(people: list[dict]) -> list[Person]:
+    person_instances = []
+    for person_data in people:
+        name = person_data["name"]
+        age = person_data["age"]
+        person = Person(name, age)
+        person_instances.append(person)
+
+    for person_data in people:
+        current_person = Person.people[person_data["name"]]
+        if "wife" in person_data and person_data["wife"]:
+            current_person.wife = Person.people[person_data["wife"]]
+        if "husband" in person_data and person_data["husband"]:
+            current_person.husband = Person.people[person_data["husband"]]
```

Comment:
The code assumes that the 'wife' or 'husband' mentioned in `person_data` exists in the `Person.people` dictionary. If a name is not found, this will raise a `KeyError`. Consider adding a check to ensure the name exists in the dictionary before attempting to access it.